### ISSUE TITLE
Node stuck on historical sync when a block contains deploy with no approvals

### ISSUE THREAD

**Issue by RitaMAllenCA** [casper-network/casper-node#3976]
It may happen that invalid deploy has been included in the chain (old binaries used to [allow that](https://github.com/casper-network/casper-node/blob/release-1.0.0/node/src/types/deploy.rs#L806-L817)). Such deploy will have the following execution results: `"error_message": "Authorization failure: not authorized.".`
When the "new" node is trying to sync this block, it'll fetch such deploy and fail to validate it, which will result in the peer being banned:
```
{"message":"banning peer","offender":"tls:d5ec..bf7c","justification":"sent a legacy deploy which failed validation (the deploy has no approvals)"}

{"message":"PeerList: no honest peers"}

```
We should update the validation so that `Fetcher<LegacyDeploy> `doesn't check approvals (the `has_valid_hash() `check should stay intact).
This behavior should ideally be covered with NCTL test when a deploy w/o approvals is sent to node 1.0.0, then there's an upgrade to 1.5.x and the new node is able to sync correctly.

**Comment by Fraser999** [comment]
As part of this ticket, we should also check the validation functionality of all fetcher types to ensure we don't have a similar issue with a different data type.

**Comment by rafal-ch** [comment]
> As part of this ticket, we should also check the validation functionality of all fetcher types to ensure we don't have a similar issue with a different data type.

@Fraser999 we currently have 9 implementations, I see no risks in other types. See summary:
| Type        | Comment       |
|-------------|---------------|
| BlockHeader | No validation |
| Block | Header and body hash validation only|
| BlockExecutionResultsOrChunk | - For chunk (so for 1.5+ data only), we check merkle proofs<br>- For "checkable" execution results (so also for 1.5+ data), we check merkle proof<br> - For "uncheckable" execution results (aka legacy) we do not validate|
| FinalitySignature | always perform crypto verification, should be compatible with pre 1.5 FSes |
|ApprovalsHashes | not available in pre 1.5, no risk of regression|
|Deploy|full validation, but only affects 1.5+ deploys|
| LegacyDeploy| changed in this PR to only validate hashes|
|SyncLeap|super fancy validation with many bells and whistles, but not available in pre 1.5 builds, so no risk|
|TrieOrChunk|no validation for trie, merkle proof validation for chunks, but also not an pre 1.5 item, so no risk|
